conda/exceptions.py

```diff
@@ -1299,7 +1291,8 @@ def print_expected_error_report(self, error_report):
             message_builder.append('')
             self.write_out('\n'.join(message_builder))
 
-    def _calculate_ask_do_upload(self):
+    def _do_upload(self, error_report) -> None:
+        """Determine whether or not to upload the error report."""
```

Comment:
I get the reason for the refactor but don't think `_do_upload` is a good name for this function, since it doesn't really handle the "determine whether or not to upload the error report" use case. What about this?
```suggestion
    def _should_upload(self, error_report) -> None:
        """Determine whether or not to upload the error report."""
```

Reply:
since it now encapsulates the entire upload process, can we just call it `_upload`?

Reply:
Yeah, that's fine, too.